compiler/rustc_hir_analysis/src/coherence/inherent_impls.rs

```diff
@@ -186,7 +186,11 @@ impl<'tcx> InherentCollect<'tcx> {
             return;
         };
 
-        let self_ty = self.tcx.type_of(item.owner_id);
+        let mut self_ty = self.tcx.type_of(item.owner_id);
+        if matches!(self_ty.kind(), ty::Alias(..)) {
```

Comment:
I guess this should either specifically check for projections or not have any check at all. The logic in rustc_privacy should be kept in sync with this

Reply:
If I remove the check, I have this panic:

```
error[E0391]: cycle detected when finding all inherent impls defined in crate
   |
   = note: ...which requires normalizing `I8<{i8::MIN}>`...
note: ...which requires evaluating type-level constant...
  --> fake-test-src-base/symbol-names/const-generics.rs:20:9
   |
LL | impl I8<{i8::MIN}> {
   |         ^^^^^^^^^
```

Also, doesn't it make sense to normalize it for type aliases too?

And final question: you mean that if we keep this check, we should have the same in privacy where I added it (just to confirm so I can update accordingly).

Reply:
> If I remove the check, I have this panic:

that's an error, and I think it can be avoided by using the `try_normalize` variant, just like at the other normalization added in this PR. The check you are doing is useless if the projection resolves to a type that would have errored without your check (as normalization is recursive/deep)

> Also, doesn't it make sense to normalize it for type aliases too?

there are no type aliases in the type system yet ;) that's your other PR

> you mean that if we keep this check, we should have the same in privacy where I added it (just to confirm so I can update accordingly).

I just mean that they should work as similarly as possible, so if both use the try variant, that seems okay.